### PR TITLE
Use pexpect and clean up handling of replwrap

### LINKED ISSUE #116: timeouts
I'm using octave_kernel, but I'm now certain there is no (easy) way to fix my problem without changing process_metakernel in Calysto. When I run a command that takes a long time to return a prompt, or any kind of text for that matter, I get the dreaded pexpect 30-second default timeout. 

A "fix" for this was committed almost a year ago (8b4429ffafbc201b4e8af5bcbe00ac647a8176f0). However, the critical part (`timeout=None`) was reverted barely a week later (`timeout=-1`, d9c975538b30061a7a6a62bedc4e8b2fc4447f88), along with changes to pexpect itself that changed the behavior of `timeout=None` to just use the default timeout of 30 seconds.

I have to admit I understand very little of this, but I suspect the author(s) of these changes consider the original bug to be only an issue when there is a continuous stream of output from a long-running process. I would ask that they consider the case when a process runs quietly for a long time. I realize this is probably an edge case, but at least for the kind of work I do with Octave, it is not all that uncommon.

Maybe there is a "fix" that at least allows the child kernel (octave_kernel in this case) to specify the timeout. As it stands, my only options are to manually modify the pexpect files, or to modify my Octave code to spit out a stream of output that prevents the timeout from triggering. Neither of these are very satisfying.

### PR DESCRIPTION
**Pull request by blink1073**
Uses pexpect 4.2 while preserving backwards compatibility.  Fixes #116.

Tested using Calysto Bash and Octave kernels to ensure backwards compatibility.

Tested on Windows.

### CODE REVIEW

**Comment by blink1073** [comment]
...and Windows proves to be darn near impossible to support as usual.

**Comment by blink1073** [comment]
@dsblank, do you mind giving this a sanity check before we merge it?

**Comment by dsblank** [comment]
I've been following along... I must admit that I have not mastered pexpect at all... but I will take a close reading to see if I see something obvious.

File: metakernel/magics/shell_magic.py [hunk 1]
```diff
@@ -47,11 +46,10 @@ def line_shell(self, *args):
...
             os.chdir(cwd)
 
-        if resp:
-            self.kernel.Print(resp)
-
-    def eval(self, cmd):
-        return self.repl.run_command(cmd, stream_handler=self.kernel.Print)
+    def eval(self, cmd, incremental=False):
+        stream_handler = self.kernel.Print if incremental else None
+        return self.repl.run_command(cmd, timeout=None,
```
Comment (dsblank) [inline metakernel/magics/shell_magic.py]: I see that eval returns something, but it is ignored above. Is that right?

**Comment by dsblank** [comment]
Looks like a real failure on Python2. But other than that, and the question I asked, I don't see anything else. Thanks for working on this!

File: metakernel/magics/shell_magic.py (hunk 1 above)
Comment (blink1073) [inline metakernel/magics/shell_magic.py]: It is only ignored when we pass `incremental=True` (where the results are handled by line for us) or we don't care about the output.

**Comment by blink1073** [comment]
Merge at will!

**Comment by dsblank** [comment]
Thanks for spending time on this... using pexpect is a useful option for many possible kernels.

**Pull request merged**